From 5fbc692b26c47bd6930450d434eea68a9ec140ee Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 26 Nov 2020 11:31:58 +0800
Subject: [PATCH 44/73] waylandsink: release frame callback when destroyed

We would use a frame callback from the surface to indicate
that last buffer is rendered, but when we destroy the surface
and that callback is not back yet, it may cause the wayland event
queue crash.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/1729>
(cherry picked from commit 6d8133e41e3e2c2f2bc65e22675a4c68a8539ac8)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 ext/wayland/gstwaylandsink.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index dc9f4f931..60f2388d3 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -282,6 +282,8 @@ gst_wayland_sink_finalize (GObject * object)
 
   GST_DEBUG_OBJECT (sink, "Finalizing the sink..");
 
+  if (sink->callback)
+    wl_callback_destroy (sink->callback);
   if (sink->last_buffer)
     gst_buffer_unref (sink->last_buffer);
   if (sink->display)
-- 
2.20.1

